services/web/server/src/simcore_service_webserver/projects/_nodes_handlers.py

```diff
@@ -469,23 +470,55 @@ async def get_project_services_access_for_gid(request: web.Request) -> web.Respo
         ]
     )
 
+    # Initialize groups to compare with everyone group ID
+    groups_to_compare = {EVERYONE_GROUP_ID}
+
+    # Get the group from the provided group ID
+    _sharing_with_group: Group | None = await get_group_from_gid(
+        app=request.app, gid=query_params.for_gid
+    )
+
+    # Check if the group exists
+    if _sharing_with_group is None:
+        msg = f"Group ID does not exist: {query_params.for_gid}"
+        raise ValueError(msg)
+
+    # Update groups to compare based on the type of sharing group
+    if _sharing_with_group.group_type == GroupTypeInModel.PRIMARY:
+        _user_id = await get_user_id_from_gid(
+            app=request.app, primary_gid=query_params.for_gid
+        )
+        _, _user_groups, _ = await list_user_groups(app=request.app, user_id=_user_id)
+        groups_to_compare.update({int(item.get("gid")) for item in _user_groups})
+        groups_to_compare.add(query_params.for_gid)
+    elif _sharing_with_group.group_type == GroupTypeInModel.STANDARD:
+        groups_to_compare = {query_params.for_gid}
+
+    # Initialize a list for inaccessible services
     inaccessible_services = []
+
+    # Check accessibility of each service
     for service in project_services_access_rights:
         service_access_rights = service.gids_with_access_rights
 
-        # Ignore services shared with everyone
-        if service_access_rights.get(EVERYONE_GROUP_ID):
-            continue
-
-        # Check if service is accessible to the provided group
-        service_access_rights_for_gid = service_access_rights.get(
-            query_params.for_gid, {}
+        # Find common groups between service access rights and groups to compare
+        _group_intersection = set(service_access_rights.keys()).intersection(
+            groups_to_compare
```

Comment:
I would name this variable user_groups_ids

Reply:
Naming: `_groups_intersection` with S improved. But the `user_groups_selection` I disagree because it is misleading. Here we need to think about the concept of groups/organizations as a project can be shared with an organization not only the user.